charts/zookeeper/templates/zookeeper.yaml

```diff
@@ -43,10 +43,22 @@ spec:
     syncLimit: {{ .Values.config.syncLimit }}
     quorumListenOnAllIPs: {{ .Values.config.quorumListenOnAllIPs }}
   {{- end }}
-  persistence:
-    reclaimPolicy: {{ .Values.persistence.reclaimPolicy }}
-    spec:
-      storageClassName: {{ .Values.persistence.storageClassName }}
-      resources:
-        requests:
-          storage: {{ .Values.persistence.volumeSize }}
+  storage:
+    {{- if eq .Values.storageType.type "ephemeral" }}
+    ephemeral:
+      enabled: {{ .Values.ephemeral.enabled }}
+      {{- if .Values.ephemeral.emptydirvolumesource }}
+      emptydirvolumesource:
+        medium: {{ .Values.ephemeral.emptydirvolumesource.medium }}
+        sizeLimit: {{ .Values.ephemeral.emptydirvolumesource.sizeLimit }}
+       {{- end }}
+    {{- end }}
+    {{- if eq .Values.storageType.type "persistence" }}
```

Comment:
This should be part of `{{- else }}` and not as part of a separate if block. Only then would this be treated as the default even when no value is provided for `storageType.type`

Reply:
Done